bug: action sheet, alert, picker, and toast do not open when isOpen is initially true

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

I want to show IonToast on page load based on some condition, e.g:

```
const {isToastOpen} = useState(true)
...
<IonToast
  message="Some message"
  isOpen={isOpen}
>
```

if `isOpen` is initially `true` then a toast does not appear. It only appears if `isOpen` is initially `false` and then changed to `true` via timeout or any other event. 

### Expected Behavior

When I set `isOpen={true}` I expect that toast appears regardless of any other state. 

### Steps to Reproduce

Minimal STR is:

```
<IonToast
  message="Some message"
  isOpen={true}
>
```

In the provided repo the toast is rendered inside `ExploreContainer` here: https://github.com/legendar/ion-input-label-issue/blob/main/src/components/ExploreContainer.tsx#L23

### Code Reproduction URL

https://github.com/legendar/ion-input-label-issue

### Ionic Info

[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package.json'

       Require stack:
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/lib/project/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/lib/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/bin/ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module '@capacitor/android/package.json'

       Require stack:
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/lib/project/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/lib/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/index.js
       - /home/***/.config/yarn/global/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI       : 7.1.1 (/home/***/.config/yarn/global/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 7.1.3

Capacitor:

   Capacitor CLI      : 5.2.1
   @capacitor/android : not installed
   @capacitor/core    : 5.2.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   NodeJS : v16.15.0 (/home/***/.nvm/versions/node/v16.15.0/bin/node)
   npm    : 8.5.5
   OS     : Linux 5.10

### Additional Information

The following component can be used as a quick workaround:

```
function InlineToast({
  isOpen,
  ...otherProps
}): JSX.Element {
  const [isOpenInternal, setIsOpenInternal] = useState(false);

  useEffect(() => {
    // setTimeout is needed because sometimes the toast shadowDom is not ready
    setTimeout(() => {
      setIsOpenInternal(isOpen);
    }, 10);
  }, [isOpen]);

  return (
    <IonToast
      isOpen={isOpenInternal}
      {...otherProps}
    />
  );
}
```

In the provided repo replace `IonToast` with `InlineToast` component and the toast will appear.

--- a/core/src/components/action-sheet/action-sheet.tsx
+++ b/core/src/components/action-sheet/action-sheet.tsx
@@ -2,6 +2,7 @@ import type { ComponentInterface, EventEmitter } from '@stencil/core';
 import { Watch, Component, Element, Event, Host, Method, Prop, h, readTask } from '@stencil/core';
 import type { Gesture } from '@utils/gesture';
 import { createButtonActiveGesture } from '@utils/gesture/button-active';
+import { raf } from '@utils/helpers';
 import {
   BACKDROP,
   createDelegateController,
@@ -318,25 +319,32 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
 
   componentDidLoad() {
     /**
-     * Do not create gesture if:
-     * 1. A gesture already exists
-     * 2. App is running in MD mode
-     * 3. A wrapper ref does not exist
+     * Only create gesture if:
+     * 1. A gesture does not already exist
+     * 2. App is running in iOS mode
+     * 3. A wrapper ref exists
+     * 4. A group ref exists
      */
     const { groupEl, wrapperEl } = this;
-    if (this.gesture || getIonMode(this) === 'md' || !wrapperEl || !groupEl) {
-      return;
+    if (!this.gesture && getIonMode(this) === 'ios' && wrapperEl && groupEl) {
+      readTask(() => {
+        const isScrollable = groupEl.scrollHeight > groupEl.clientHeight;
+        if (!isScrollable) {
+          this.gesture = createButtonActiveGesture(wrapperEl, (refEl: HTMLElement) =>
+            refEl.classList.contains('action-sheet-button')
+          );
+          this.gesture.enable(true);
+        }
+      });
     }
 
-    readTask(() => {
-      const isScrollable = groupEl.scrollHeight > groupEl.clientHeight;
-      if (!isScrollable) {
-        this.gesture = createButtonActiveGesture(wrapperEl, (refEl: HTMLElement) =>
-          refEl.classList.contains('action-sheet-button')
-        );
-        this.gesture.enable(true);
-      }
-    });
+    /**
+     * If action sheet was rendered with isOpen="true"
+     * then we should open action sheet immediately.
+     */
+    if (this.isOpen === true) {
+      raf(() => this.present());
+    }
   }
 
   render() {

--- a/core/src/components/action-sheet/test/is-open/action-sheet.e2e.ts
+++ b/core/src/components/action-sheet/test/is-open/action-sheet.e2e.ts
@@ -30,5 +30,10 @@ configs({ directions: ['ltr'], modes: ['ios'] }).forEach(({ config, title }) => 
 
       await expect(actionSheet).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-action-sheet is-open="true"></ion-action-sheet>', config);
+      await expect(page.locator('ion-action-sheet')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/alert/alert.tsx
+++ b/core/src/components/alert/alert.tsx
@@ -347,7 +347,7 @@ export class Alert implements ComponentInterface, OverlayInterface {
 
   componentDidLoad() {
     /**
-     * Only  create gesture if:
+     * Only create gesture if:
      * 1. A gesture does not already exist
      * 2. App is running in iOS mode
      * 3. A wrapper ref exists
@@ -360,8 +360,8 @@ export class Alert implements ComponentInterface, OverlayInterface {
     }
 
     /**
-     * If toast was rendered with isOpen="true"
-     * then we should open toast immediately.
+     * If alert was rendered with isOpen="true"
+     * then we should open alert immediately.
      */
     if (this.isOpen === true) {
       raf(() => this.present());

--- a/core/src/components/alert/alert.tsx
+++ b/core/src/components/alert/alert.tsx
@@ -3,6 +3,7 @@ import { Component, Element, Event, Host, Listen, Method, Prop, Watch, forceUpda
 import { ENABLE_HTML_CONTENT_DEFAULT } from '@utils/config';
 import type { Gesture } from '@utils/gesture';
 import { createButtonActiveGesture } from '@utils/gesture/button-active';
+import { raf } from '@utils/helpers';
 import {
   createDelegateController,
   createTriggerController,
@@ -346,19 +347,25 @@ export class Alert implements ComponentInterface, OverlayInterface {
 
   componentDidLoad() {
     /**
-     * Do not create gesture if:
-     * 1. A gesture already exists
-     * 2. App is running in MD mode
-     * 3. A wrapper ref does not exist
+     * Only  create gesture if:
+     * 1. A gesture does not already exist
+     * 2. App is running in iOS mode
+     * 3. A wrapper ref exists
      */
-    if (this.gesture || getIonMode(this) === 'md' || !this.wrapperEl) {
-      return;
+    if (!this.gesture && getIonMode(this) === 'ios' && this.wrapperEl) {
+      this.gesture = createButtonActiveGesture(this.wrapperEl, (refEl: HTMLElement) =>
+        refEl.classList.contains('alert-button')
+      );
+      this.gesture.enable(true);
     }
 
-    this.gesture = createButtonActiveGesture(this.wrapperEl, (refEl: HTMLElement) =>
-      refEl.classList.contains('alert-button')
-    );
-    this.gesture.enable(true);
+    /**
+     * If toast was rendered with isOpen="true"
+     * then we should open toast immediately.
+     */
+    if (this.isOpen === true) {
+      raf(() => this.present());
+    }
   }
 
   /**

--- a/core/src/components/alert/test/is-open/alert.e2e.ts
+++ b/core/src/components/alert/test/is-open/alert.e2e.ts
@@ -29,5 +29,10 @@ configs({ directions: ['ltr'], modes: ['ios'] }).forEach(({ config, title }) => 
       await ionAlertDidDismiss.next();
       await expect(alert).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-alert is-open="true"></ion-alert>', config);
+      await expect(page.locator('ion-alert')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/loading/test/is-open/loading.e2e.ts
+++ b/core/src/components/loading/test/is-open/loading.e2e.ts
@@ -23,5 +23,10 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => 
       await ionLoadingDidDismiss.next();
       await expect(loading).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-loading is-open="true"></ion-loading>', config);
+      await expect(page.locator('ion-loading')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/modal/test/is-open/modal.e2e.ts
+++ b/core/src/components/modal/test/is-open/modal.e2e.ts
@@ -20,5 +20,10 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => 
       await ionModalDidDismiss.next();
       await expect(modal).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-modal is-open="true"></ion-modal>', config);
+      await expect(page.locator('ion-modal')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/picker/picker.tsx
+++ b/core/src/components/picker/picker.tsx
@@ -1,5 +1,6 @@
 import type { ComponentInterface, EventEmitter } from '@stencil/core';
 import { Component, Element, Event, Host, Method, Prop, State, Watch, h } from '@stencil/core';
+import { raf } from '@utils/helpers';
 import {
   createDelegateController,
   createTriggerController,
@@ -197,6 +198,16 @@ export class Picker implements ComponentInterface, OverlayInterface {
 
   componentWillLoad() {
     setOverlayId(this.el);
+  }
+
+  componentDidLoad() {
+    /**
+     * If picker was rendered with isOpen="true"
+     * then we should open picker immediately.
+     */
+    if (this.isOpen === true) {
+      raf(() => this.present());
+    }
   }
 
   /**

--- a/core/src/components/picker/test/is-open/picker.e2e.ts
+++ b/core/src/components/picker/test/is-open/picker.e2e.ts
@@ -23,5 +23,10 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => 
       await ionPickerDidDismiss.next();
       await expect(picker).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-picker is-open="true"></ion-picker>', config);
+      await expect(page.locator('ion-picker')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/popover/test/is-open/popover.e2e.ts
+++ b/core/src/components/popover/test/is-open/popover.e2e.ts
@@ -1,0 +1,11 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => {
+  test.describe(title('popover: isOpen'), () => {
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-popover is-open="true"></ion-popover>', config);
+      await expect(page.locator('ion-popover')).toBeVisible();
+    });
+  });
+});

--- a/core/src/components/toast/test/is-open/toast.e2e.ts
+++ b/core/src/components/toast/test/is-open/toast.e2e.ts
@@ -24,5 +24,10 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => 
       await ionToastDidDismiss.next();
       await expect(toast).toBeHidden();
     });
+
+    test('should open if isOpen is true on load', async ({ page }) => {
+      await page.setContent('<ion-toast is-open="true"></ion-toast>', config);
+      await expect(page.locator('ion-toast')).toBeVisible();
+    });
   });
 });

--- a/core/src/components/toast/toast.tsx
+++ b/core/src/components/toast/toast.tsx
@@ -1,6 +1,7 @@
 import type { ComponentInterface, EventEmitter } from '@stencil/core';
 import { State, Watch, Component, Element, Event, h, Host, Method, Prop } from '@stencil/core';
 import { ENABLE_HTML_CONTENT_DEFAULT } from '@utils/config';
+import { raf } from '@utils/helpers';
 import { printIonWarning } from '@utils/logging';
 import {
   createDelegateController,
@@ -251,6 +252,16 @@ export class Toast implements ComponentInterface, OverlayInterface {
 
   componentWillLoad() {
     setOverlayId(this.el);
+  }
+
+  componentDidLoad() {
+    /**
+     * If toast was rendered with isOpen="true"
+     * then we should open toast immediately.
+     */
+    if (this.isOpen === true) {
+      raf(() => this.present());
+    }
   }
 
   /**

